src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -158,6 +160,45 @@ export function prefillTransformer(pages, formData, metadata) {
   };
 }
 
+/**
+ * Transforms the related disabilities object into an array of strings
+ * @param {Object} object - The object with dynamically generated property names
+ * @return {Array} - An array of the property names with truthy values
+ */
+export function transformRelatedDisabilities(object) {
+  return Object.keys(object).filter(key => object[key]);
+}
+
+export function transform(formConfig, form) {
+  // Have to do this first or it messes up the results from transformRelatedDisabilities for some reason.
+  // The transformForSubmit's JSON.stringify transformer doesn't remove deeply empty objects, so we call
+  //  it here to remove reservesNationalGuardService if it's deeply empty.
+  const clonedData = removeDeeplyEmptyObjects(
+    JSON.parse(transformForSubmit(formConfig, form)),
+  );
+
+  // Transform the related disabilities lists into an array of strings
+  if (clonedData.vaTreatmentFacilities) {
+    const newVAFacilities = clonedData.vaTreatmentFacilities.map(facility =>
+      _.set(
+        'relatedDisabilities',
+        transformRelatedDisabilities(facility.relatedDisabilities),
+        facility,
+      ),
+    );
+    clonedData.vaTreatmentFacilities = newVAFacilities;
+  }
+  if (clonedData['view:isPOW']) {
+    clonedData['view:isPOW'] = transformRelatedDisabilities(
+      _.get('view:isPOW.powDisabilities', clonedData),
+    );
+  }
+
+  return JSON.stringify({
+    form526: JSON.stringify(clonedData),
```

Comment:
Would be nice if we didn't have to stringify, parse, and then re-stringify the form data, but I guess that depends on what the deal is with the `transformRelatedDisabilities` bit you commented on earlier.

Reply:
Yup. Exactly. So having the `transformForSubmit` at the end like I originally had it was messing up the `relatedDisabilities` array, so I ended up putting it at the beginning of this function. `transformForSubmit` returns a JSON-stringified object, so in order to continue massaging the data, I had to parse it...only to stringify it again right at the end. :disappointed: